widget/embedded/src/hooks/useSwapInput.ts

```diff
@@ -113,15 +122,29 @@ export function useSwapInput(): UseSwapInput {
       fetchQuote(requestBody)
         .then((res) => {
           setLoading(false);
-          setQuote(res);
-          throwErrorIfResponseIsNotValid(res);
-          const quoteWarning = generateQuoteWarnings(res, {
-            fromToken,
-            toToken,
-            userSlippage,
-            tokens,
-          });
+          setQuotes(res);
+          const sortQuotes = sortRoutesBy(sortStrategy, res.results);
+          const quote = sortQuotes.length ? sortQuotes[0] : null;
+          const sortedQuoteTags = sortTags(quote?.tags || []);
+          setSelectedQuote(quote);
+          setSelectedTag(sortedQuoteTags[0]);
```

Comment:
This approach lacks future-proofing. In scenarios where a route has multiple tags such as 'fastest' and 'lowest fee,' and the sorting strategy is based on speed or fee, there is no assurance that the correct tag will be returned.